server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -1349,7 +1353,7 @@ public ShardAllocationDecision decideShardAllocation(ShardRouting shard, Routing
     }
 
     private static DesiredBalanceComputer createDesiredBalanceComputer(ShardsAllocator allocator) {
-        return new DesiredBalanceComputer(createBuiltInClusterSettings(), mock(ThreadPool.class), allocator);
+        return new DesiredBalanceComputer(createBuiltInClusterSettings(), mock(ThreadPool.class)::relativeTimeInMillis, allocator);
```

Comment:
It looks like we no longer need mock here
```suggestion
        return new DesiredBalanceComputer(createBuiltInClusterSettings(), () -> 0L, allocator);
```